etc/travis/DeploymentTester.py

```diff
@@ -0,0 +1,115 @@
+        print("(logged in)")
+
+    def create_resource_group(self):
+        print('\nCreating group "{}" on "{}"...'.format(self.config.resource_group, self.config.location))
+        self.resource_client.resource_groups.create_or_update(self.config.resource_group,
+                                                              {'location': self.config.location})
+        print('(created)')
+
+    def validate(self):
+        print('\nValidating template...')
+
+        validation = self.resource_client.deployments.validate(self.config.resource_group,
+                                                               self.config.deployment_name,
+                                                               self.config.deployment_properties)
+        if validation.error is not None:
+            print("*** VALIDATION FAILED ***")
+            print(validation.error.message)
+            sys.exit(1)
+
+        print("(valid)")
+
+    def deploy(self):
+        print('\nDeploying template, feel free to take a nap...')
+        deployment = self.resource_client.deployments.create_or_update(self.config.resource_group,
+                                                                       self.config.deployment_name,
+                                                                       self.config.deployment_properties)
+        """:type : msrestazure.azure_operation.AzureOperationPoller"""
+        started = time.time()
+        while not deployment.done():
+            print('... after {} still "{}" ...'.format(self.elapsed(started), deployment.status()))
+            deployment.wait(60)
+        print("WAKE UP! After {} we finally got status {}.".format(self.elapsed(started), deployment.status()))
+
+        print("Checking deployment response...")
+        properties = deployment.result(0).properties
+        if properties.provisioning_state != 'Succeeded':
+            print("*** DEPLOY FAILED ***")
+            print('Provisioning state: ' + properties.provisioning_state)
+            sys.exit(1)
+        self.load_deployment_outputs(properties.outputs)
+        print("(success)")
+
+    def load_deployment_outputs(self, outputs):
+        self.deployment = {}
+        for key, value in outputs.items():
+            self.deployment[key] = value['value']
+            print("- Found: " + key)
+
+    def moodle_smoke_test(self):
+        print("Moodle Smoke Test...")
+        url = 'https://' + self.deployment['siteURL']
+        curl = Curl()
+        curl.setopt(pycurl.URL, url)
+        curl.setopt(pycurl.SSL_VERIFYPEER, False)
+        curl.setopt(pycurl.WRITEFUNCTION, lambda x: None)
+        curl.perform()
+        status = curl.getinfo(pycurl.HTTP_CODE)
+        if status != 200:
+            print("*** DEPLOY FAILED ***")
+            print('HTTP Status Code: ' + status)
+            sys.exit(1)
+        print('(ok: {})'.format(status))
```

Comment:
Can we try just one more here -- Login to the site as the admin user and see if works?

Reply:
If I perform a login as admin, it would be interesting to go to the notifications and check if there is any error detected by Moodle as well. I'll see if its reasonable to do that and ping you back.

For now I could probably do it only with curl. Are we planning to add further tests later -- would it have any value to use selenium in the future?

Reply:
Probably not much to add further tests in the future, but I wanted to make sure at this point that the database is also correctly installed/deployed/configured. A successful Moodle admin login will show that to us, so that's why I'm asking. Using curl should be just fine. We won't be using a too like selenium in the future, at least no plan currently.

Reply:
A get in the index by itself does test the DB conectivity and many subsystems of Moodle because they are required even for a simple "login page" rendering (there are authentication settings for example stored in the database).

In any case, I implemented a CURL (with follow redirects) to ensure Admin could login and gets to its front page, that gives some extra assertions that the framework is not corrupt. See https://github.com/Azure/Moodle/pull/91/files#diff-0800a30435a8e602ff7bf8df45fc6e09R131

Reply:
Great to know that hitting the index.php does perform important operations. Thanks for your education. Also thanks much for your admin login check as well!